src/Lookup/InProcessCachingDataTypeLookup.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\PropertyId;
+
+/**
+ * PropertyDataTypeLookup that provides in-process caching.
+ *
+ * @since 3.1
+ *
+ * @licence GNU GPL v2+
+ * @author Katie Filbert < aude.wiki@gmail.com >
+ */
+class InProcessCachingDataTypeLookup implements PropertyDataTypeLookup {
```

Comment:
I thought of using HashBagOStuff, but then we would have to put the CachingPropertyDataTypeLookup in WikibaseLib, since we then have a dependency on core.